modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/viewmodel/StorageSettingsViewModel.kt

```diff
@@ -158,6 +166,48 @@ class StorageSettingsViewModel
         analyticsTracker.track(AnalyticsEvent.SETTINGS_STORAGE_CLEAR_DOWNLOAD_CACHE)
     }
 
+    fun fixDownloadedFiles() {
+        analyticsTracker.track(AnalyticsEvent.SETTINGS_STORAGE_FIX_DOWNLOADED_FILES)
+        fixDownloadsJob?.cancel()
+        fixDownloadsJob = viewModelScope.launch {
+            val episodeCount = episodeManager.countEpisodes()
+            mutableState.update { state ->
+                state.copy(fixDownloadsState = State.FixDownloadsState(isFixing = true, episodeCount = episodeCount))
+            }
+
+            repeat(episodeCount / FIX_EPISODES_LIMIT + 1) { iteration ->
+                val offset = iteration * FIX_EPISODES_LIMIT
+                val episodes = episodeManager.getAllPodcastEpisodes(FIX_EPISODES_LIMIT, offset)
+                episodes.forEachIndexed { index, episode ->
```

Comment:
We don't have to do it for this PR but I wonder if it would be nice to have a paging method we could use for other code in the future. Something like:
```
episodeManager.getPodcastEpisodesPaged() { episode -> 
```

Reply:
I like the idea. Though this type of paging would be mainly for business logic and not for UI were it integrates mainly through `PagedList`. I'll add it to this PR.

Reply:
954691438a161dc570975123e08709ef501715fa